libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -284,130 +262,123 @@ func getContainersInPod(pod *kubernetes.Pod) ([]kubernetes.Container, []kubernet
 	}
 	statuses = append(statuses, pod.Status.EphemeralContainerStatuses...)
 
-	return containers, statuses
+	containersInPod := make([]*containerInPod, len(containers))
+	for i, c := range containers {
+		cp := containerInPod{spec: c}
+		containersInPod[i] = &cp
+
+		// This nested loop can be a problem in pods with hundreds of containers, does it exist?
```

Comment:
Refactored to use a map again to avoid the worst case scenario, and for loops to avoid generating so many temporary objects, as proposed in https://github.com/elastic/beats/pull/24742#discussion_r611388219.

Sorting the arrays is actually a good suggestion but complicates things a little bit, and can be also too much for the common use case.

We can revisit this function in a follow up if we see that needs to be improved.

Reply:
👍🏼